client/components/category-pill-navigation/index.tsx

```diff
@@ -36,6 +38,7 @@ export const CategoryPillNavigation = ( {
 	const [ showRightArrow, setShowRightArrow ] = useState( false );
 	const listInnerRef = useRef< HTMLDivElement | null >( null );
 	const isRtl = useRtl();
+	const { recordPatternsEvent } = useRecordPatternsEvent();
```

Comment:
We can't use this approach, since category-pill-navigation supposed to be used on different pages, so we can't make any dependency on the specific page.

The best approach that I see is:
1) Move `const { recordPatternsEvent } = useRecordPatternsEvent();` outside to the parent component
2) Inside CategoryPillNavigation we can expect an argument e.g. `onClickPill?: (id: string) => void`, so the parent will provide the next function `(id: string) => recordPatternsEvent( 'calypso_pattern_library_filter', { category: id } )`
3) Inside the PillNavigation we will do `onClick={ () => onClickPill(button.id)` and `onClick={ () => onClickPill(category.id)`

Reply:
Thanks for the proposal. I've implemented it and it's working as expected!